Assets/Plugins/Editor/Uplift/Schemas/Repositories/GithubRepository.cs

```diff
@@ -0,0 +1,128 @@
+// --- BEGIN LICENSE BLOCK ---
+/*
+ * Copyright (c) 2017-present WeWantToKnow AS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+// --- END LICENSE BLOCK ---
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Linq;
+using System.IO;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEngine;
+using Uplift.Common;
+using Uplift.GitHubModule;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Uplift.Schemas
+{
+    public partial class GithubRepository : Repository
+    {
+        private GitHubRelease release;
```

Comment:
I don't remember the lifecycle here, when do we reinitialize those instances? Given that we keep state here, would that prevent us from refreshing?

Reply:
The GithubRepository object is kept in the package list instance. When we want to refresh, the repositories in the PackageList are refreshed, so this object will be garbage collected. So it does not prevent us from refreshing in theory. But I'll check that practice matches theory!

Reply:
Verified! The rationale being keeping it in a object is that without it we would do LOTS of call to GitHub, which had a significative impact on performance